src/language/utils.ts

```diff
@@ -73,6 +73,19 @@ export function scanAllTokens(scanner: ts.Scanner, callback: (s: ts.Scanner) =>
     }
 }
 
+export function scanAllTokensWithSkip(scanner: ts.Scanner, skipMap: Map<number, number>, callback: (s: ts.Scanner) => void) {
```

Comment:
can you please add JSDoc comments to these two new util functions explaining why they are useful and in which cases they are better than the SkippableTokenAwareRuleWalker?